src/Forms/TextareaField.php

```diff
@@ -45,19 +45,29 @@ class TextareaField extends FormField
      */
     protected $cols = 20;
 
+    /**
+     * @var int
+     */
+    protected $maxLength;
+
     /**
      * Set textarea specific schema data
      */
     public function getSchemaDataDefaults()
     {
         $data = parent::getSchemaDataDefaults();
-        
+
         $data['data']['rows'] = $this->getRows();
         $data['data']['columns'] = $this->getColumns();
-        
+
+        $maxLength = $this->getMaxLength();
+        if ($maxLength) {
+            $data['maxLength'] = $maxLength;
```

Comment:
Top level form schema keys should only be for properties that apply to all fields; maxLength only makes sense to text-based fields.

This should go into `$data['attributes']['maxLength']` or `$data['data']['maxLength']`